include/SFML/Graphics/Color.inl

```diff
@@ -0,0 +1,149 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2021 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+
+////////////////////////////////////////////////////////////
+constexpr Color::Color() :
+r(0),
+g(0),
+b(0),
+a(255)
+{
+
+}
+
+
+////////////////////////////////////////////////////////////
+constexpr Color::Color(Uint8 red, Uint8 green, Uint8 blue, Uint8 alpha) :
+r(red),
+g(green),
+b(blue),
+a(alpha)
+{
+
+}
+
+
+////////////////////////////////////////////////////////////
+constexpr Color::Color(Uint32 color) :
+r(static_cast<Uint8>((color & 0xff000000) >> 24)),
+g(static_cast<Uint8>((color & 0x00ff0000) >> 16)),
+b((color & 0x0000ff00) >> 8),
+a((color & 0x000000ff) >> 0)
```

Comment:
There should be integral promotion for all 4 of them, why cast some but not all?
The code would probably be more readable 😉 

Also, not sure if `>> 0` is that helpful, we also don't do the opposite in `toInteger()`.

Reply:
@Bromeon Done! :)